lib/screens/v2/widget_instance/subway_status.ex

```diff
@@ -62,54 +100,127 @@ defmodule Screens.V2.WidgetInstance.SubwayStatus do
+  def serialize_one_row_for_all_routes(grouped_alerts) do
+    %{
+      blue: serialize_single_alert_row_for_route(grouped_alerts, "Blue"),
+      orange: serialize_single_alert_row_for_route(grouped_alerts, "Orange"),
+      red: serialize_single_alert_row_for_route(grouped_alerts, "Red"),
+      green: serialize_green_line(grouped_alerts)
+    }
+  end
+
+  # At most 1 alert on any route
+  def serialize_routes_zero_or_one_alert(grouped_alerts) do
+    %{
+      blue: serialize_single_alert_row_for_route(grouped_alerts, "Blue"),
+      orange: serialize_single_alert_row_for_route(grouped_alerts, "Orange"),
+      red: serialize_single_alert_row_for_route(grouped_alerts, "Red"),
+      green: serialize_green_line(grouped_alerts)
+    }
+  end
+
+  # More than 1 alert on any one route
+  def serialize_routes_multiple_alerts(grouped_alerts) do
+    routes_with_alerts = Map.keys(grouped_alerts)
+    total_alerts = grouped_alerts |> Enum.flat_map(&elem(&1, 1)) |> length()
+
+    if Enum.any?(routes_with_alerts, &String.starts_with?(&1, "Green")) do
+      # Collapse all non-GL routes and display as many GL alerts as possible.
+      %{
+        blue: serialize_single_alert_row_for_route(grouped_alerts, "Blue"),
+        orange: serialize_single_alert_row_for_route(grouped_alerts, "Orange"),
+        red: serialize_single_alert_row_for_route(grouped_alerts, "Red"),
+        green: serialize_green_line(grouped_alerts)
+      }
+    else
+      if length(routes_with_alerts) == 1 and total_alerts == 2 do
+        # Show both alerts in two rows
+        %{
+          blue: serialize_multiple_alert_rows_for_route(grouped_alerts, "Blue"),
+          orange: serialize_multiple_alert_rows_for_route(grouped_alerts, "Orange"),
+          red: serialize_multiple_alert_rows_for_route(grouped_alerts, "Red"),
+          green: serialize_multiple_alert_rows_for_route(grouped_alerts, "Green")
+        }
+      else
+        # Collapse all routes
+        serialize_one_row_for_all_routes(grouped_alerts)
+      end
+    end
+  end
+
+  # Only executed if one non-GL route has exactly 2 alerts
+  def serialize_multiple_alert_rows_for_route(grouped_alerts, route_id) do
+    alerts = Map.get(grouped_alerts, route_id)
+
+    alert_rows =
+      if is_nil(alerts) or alerts == [] do
+        [%{route_pill: serialize_route_pill(route_id), status: "Normal Service"}]
+      else
+        [alert1, alert2] = alerts
+
+        [
+          Map.merge(
+            %{route_pill: serialize_route_pill(route_id)},
+            serialize_alert(alert1, route_id)
+          ),
+          serialize_alert(alert2, route_id)
+        ]
+      end
+
+    %{
+      type: :contracted,
+      alerts: alert_rows
+    }
+  end
+
+  # Only executed when route displays one status.
+  def serialize_single_alert_row_for_route(grouped_alerts, route_id) do
     alerts = Map.get(grouped_alerts, route_id)
+    total_alerts = grouped_alerts |> Enum.flat_map(&elem(&1, 1)) |> Enum.uniq() |> length()
```

Comment:
Good catch. GL unfortunately requires us to use `uniq_by` so that we don't count multiple branch alerts twice. This will allow us to count a Gov't Ctr closure as 2 alerts instead of 5.